### PR TITLE
feat: Update Navbar functionality

### LINKED ISSUE #474: [Intermediate]:  Update Navbar functionality
### 🧑‍💻 Intermediate Issue

Welcome! This is an **[Intermediate Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22intermediate%22%20repo%3Ahiero-ledger%2Fhiero-website)** touching core website architecture.

🏁 **When this issue is complete, you will have:**

✅ Researched across multiple files and modules
✅ Owned implementation decisions
✅ Written thorough, meaningful tests
✅ Delivered a clean, review-ready pull request

### 🐞 Problem Description

With the new redesign of the website we want to implement new functionality. 

Currently our navbar is present when contributors scroll. 

Our redesign changes this feature to hidden when users being scrolling on the site. 

### 💡 Expected Solution

Change the scroll feature from showing the navbar to hiding the navbar. 

PUSH WORK TO websiteRedesign BRANCH

### 🔍 Background Research

Use this site to to review the functionality of how we expect the navbar to appear. 

https://hedera.com/

Design File:
https://www.figma.com/design/YndiSK4Z4P2yXPxlsj3UHU/Hiero-Website-Design-Darshit?node-id=1-2&t=Ewl5C6gzh6PZIPo9-0

Markdown Specification: 
https://github.com/hiero-ledger/hiero-website/blob/main/docs/design-specs.md

**Blog & Content Workflow:**
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md) — covers the current posting flow, front matter format, and content conventions

**Project Conventions:**
- Components live in `src/components/` — follow existing patterns for props, naming, and file structure
- Pages are under `src/app/` using the Next.js App Router
- Content (blog posts, etc.) lives in `content/`
- Styling uses [Tailwind CSS](https://tailwindcss.com/docs)

### 🛠️ Implementation Notes

Implement new feature to automatically hide and show navbar depending on if users are scrolling up or down. 

### 🔬 Technical Domains

      

- [ ] **Intermediate to Advanced Web Development** (React components, hooks, state management, Next.js routing)
- [ ] **Component Architecture** (reusable components, props design, composition patterns)
- [ ] **Styling & Responsiveness** ([Tailwind CSS](https://tailwindcss.com/docs), responsive design, accessibility)
- [ ] **Performance** (image optimization, lazy loading, Core Web Vitals)
- [ ] **Testing** (unit, integration, snapshot, accessibility testing)

### 🧠 Intermediate Contributors — Prerequisites & Expectations

> [!CAUTION]
> **Intermediate issues are high-risk. We expect more than just a 'working solution' and will recommend beginner issues if the PR does not meet these standards.**

### 🏁 Concrete Prerequisites
- **Web Development Skills:** Intermediate or advanced React/TypeScript/Next.js skills.
- **Expertise:** Strong understanding of files related to this issue (research before claiming!).
- **Proven History:** Successfully completed **≥ 5 beginner issues** in this repo.

> [!NOTE]
> **CI/CD Exception:** For issues focused on **GitHub Actions / Workflows**, the repo-specific thresholds above may be waived if the contributor demonstrates intermediate-level proficiency in CI/CD.

If this feels like too big a step, that is completely fine — try a [Beginner Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22beginner%22%20repo%3Ahiero-ledger%2Fhiero-website) first or a different **[Intermediate Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22intermediate%22%20repo%3Ahiero-ledger%2Fhiero-website)**. You can always come back when you are ready.

### ⚠️ AI Usage Policy

- Using AI to generate code for Intermediate issues is **strictly discouraged**
- Using AI as the main source of research is **strictly discouraged**, refer to React, Next.js, and TypeScript documentation.

### ⏱️ Timeline & Workflow
- **Typical time:** ~2 weeks / ~25 hours.
- 🔴 Completing an intermediate issue in 1–3 days is a **red flag**.

> [!TIP]
> **Suggested:** share your proposed implementation approach as a comment before writing code to get early feedback and avoid wasted effort.

### 🧪 Testing Requirements

How you test depends on the type of change.

> [!IMPORTANT]
> At the intermediate level, **testing is a major component**.
> Each component or function you implement should be verified. Tests should cover happy paths,
> edge cases, and error handling.

**Component or source code changes (e.g. in `src/`):**
- Write unit tests covering happy path, edge cases, and error handling
- Run tests locally: `pnpm test`
- Verify the dev server renders correctly: `pnpm dev`
- Check consistency with similar components already in the codebase — use the same patterns

**GitHub Actions / workflow changes (e.g. in `.github/`):** [Guide](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/github-action-workflows.md)
- Test by merging to your fork's `main` and simulating the scenario
- Create test issues or PRs as evidence and link them in your PR
- Include screenshots of workflow runs where applicable

**Content, styling, or layout changes:**
- Run the dev server and confirm visual output matches expected behavior
- Compare your changes with similar existing pages for consistency
- Check responsive behavior and accessibility

### 🛡️ Quality & Review Standards

Intermediate PRs must be **"working, maintainable, and aligned with project architecture."**

  1. **Working:** The implementation must solve the problem and meet the acceptance criteria.
  2. **Maintainable:** Code should be clear and concise enough for others to understand and debug without your assistance.
  3. **Architectural Alignment:** The solution should fit well with existing component patterns and project conventions.
  4. **Backward Compatibility:** Existing pages and components must not break. 
  5. **Comprehensive Testing:** Must include unit tests covering all new logic paths, edge cases, and failure modes. AI-generated tests based on AI-generated code are grounds for immediate rejection.

**⚠️ Breaking changes**
- Before changing any component API, prop interface, or public-facing layout — stop and check
- If a breaking change is unavoidable: get explicit maintainer approval **before** implementing
- All existing tests should pass as-is.

**🤖 AI notes**
- AI often has out-dated real-world knowledge: it will not be able to identify current packages.
- AI has gaps or incorrect understanding of repo-specific patterns, logic, and component structures.

### ✅ PR Quality Checklist

Before opening your PR, confirm:

- [ ] I have spent the majority of my time researching the problem and solution space extensively, including reviewing relevant code, documentation, and external resources.
- [ ] My implementation works but is also of high quality - including maintainability, readability, and architectural fit.
- [ ] I have checked for breaking changes - no existing pages or components regress.
- [ ] The design fits with the current website's architectural approaches.
- [ ] Every line of code is personally understood and explainable.

Before requesting a review, confirm:
- [ ] I have reviewed the diff line by line.
- [ ] My implementation fully addresses the problem described above.
- [ ] I did not modify files unrelated to this issue
- [ ] Clean git history — no rebase artifacts, merge commits, or unrelated files
- [ ] My commits are signed: `git commit -S -s -m "chore: description"` — [Signing guide](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/signing.md)
- [ ] I have run the following commands locally and **all pass** (CI will automatically fail your PR if any of these fail):
  - `pnpm format:check` — checks code formatting
  - `pnpm lint` — runs the linter (**the CI lint check will block your PR if this fails**)
  - `pnpm test` — runs the test suite
  - `pnpm build` — ensures the project builds successfully
  - [Testing guide](https://github.com/hiero-ledger/hiero-website/blob/main/docs/06-testing-and-quality-checks.md)
- [ ] I have included appropriate tests and all CI checks pass.
- [ ] Double and triple check — intermediate PRs are time-consuming to review

### 📚 Resources & Support

**🆘 Stuck?**
> [!TIP]
> **Comment on this issue:** and describe what you have tried. A maintainer will respond.

**References:**
- [Next.js Documentation](https://nextjs.org/docs)
- [React Documentation](https://react.dev)
- [Tailwind CSS Documentation](https://tailwindcss.com/docs)
- [Vitest Documentation](https://vitest.dev)
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md)
- [Community Calls](https://zoom-lfx.platform.linuxfoundation.org/meetings/hiero?view=week)
- [Discord](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/discord.md)

### LINKED ISSUE #475: [Intermediate]:  Expand Navbar Icons
### 🧑‍💻 Intermediate Issue

Welcome! This is an **[Intermediate Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22intermediate%22%20repo%3Ahiero-ledger%2Fhiero-website)** touching core website architecture.

🏁 **When this issue is complete, you will have:**

✅ Researched across multiple files and modules
✅ Owned implementation decisions
✅ Written thorough, meaningful tests
✅ Delivered a clean, review-ready pull request

### 🐞 Problem Description

Currently we have icons for discord and GitHub. 

While this is good our redesign aims to expand these icons and make them more noticeable to contributors.  

### 💡 Expected Solution

Redesign the icons to make them present as buttons rather then standalone images. 

PUSH WORK TO websiteRedesign BRANCH

### 🔍 Background Research

Follow this design doc to see how this should appear

https://www.figma.com/design/YndiSK4Z4P2yXPxlsj3UHU/Hiero-Website-Design-Darshit?node-id=1-2&t=Ewl5C6gzh6PZIPo9-0

Markdown Specification: 
https://github.com/hiero-ledger/hiero-website/blob/main/docs/design-specs.md

**Blog & Content Workflow:**
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md) — covers the current posting flow, front matter format, and content conventions

**Project Conventions:**
- Components live in `src/components/` — follow existing patterns for props, naming, and file structure
- Pages are under `src/app/` using the Next.js App Router
- Content (blog posts, etc.) lives in `content/`
- Styling uses [Tailwind CSS](https://tailwindcss.com/docs)

### 🛠️ Implementation Notes

Implement clickable buttons that direct contributors to the Hiero GitHub and Discord pages. 

### 🔬 Technical Domains

      

- [ ] **Intermediate to Advanced Web Development** (React components, hooks, state management, Next.js routing)
- [ ] **Component Architecture** (reusable components, props design, composition patterns)
- [ ] **Styling & Responsiveness** ([Tailwind CSS](https://tailwindcss.com/docs), responsive design, accessibility)
- [ ] **Performance** (image optimization, lazy loading, Core Web Vitals)
- [ ] **Testing** (unit, integration, snapshot, accessibility testing)

### 🧠 Intermediate Contributors — Prerequisites & Expectations

> [!CAUTION]
> **Intermediate issues are high-risk. We expect more than just a 'working solution' and will recommend beginner issues if the PR does not meet these standards.**

### 🏁 Concrete Prerequisites
- **Web Development Skills:** Intermediate or advanced React/TypeScript/Next.js skills.
- **Expertise:** Strong understanding of files related to this issue (research before claiming!).
- **Proven History:** Successfully completed **≥ 5 beginner issues** in this repo.

> [!NOTE]
> **CI/CD Exception:** For issues focused on **GitHub Actions / Workflows**, the repo-specific thresholds above may be waived if the contributor demonstrates intermediate-level proficiency in CI/CD.

If this feels like too big a step, that is completely fine — try a [Beginner Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22beginner%22%20repo%3Ahiero-ledger%2Fhiero-website) first or a different **[Intermediate Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22intermediate%22%20repo%3Ahiero-ledger%2Fhiero-website)**. You can always come back when you are ready.

### ⚠️ AI Usage Policy

- Using AI to generate code for Intermediate issues is **strictly discouraged**
- Using AI as the main source of research is **strictly discouraged**, refer to React, Next.js, and TypeScript documentation.

### ⏱️ Timeline & Workflow
- **Typical time:** ~2 weeks / ~25 hours.
- 🔴 Completing an intermediate issue in 1–3 days is a **red flag**.

> [!TIP]
> **Suggested:** share your proposed implementation approach as a comment before writing code to get early feedback and avoid wasted effort.

### 🧪 Testing Requirements

How you test depends on the type of change.

> [!IMPORTANT]
> At the intermediate level, **testing is a major component**.
> Each component or function you implement should be verified. Tests should cover happy paths,
> edge cases, and error handling.

**Component or source code changes (e.g. in `src/`):**
- Write unit tests covering happy path, edge cases, and error handling
- Run tests locally: `pnpm test`
- Verify the dev server renders correctly: `pnpm dev`
- Check consistency with similar components already in the codebase — use the same patterns

**GitHub Actions / workflow changes (e.g. in `.github/`):** [Guide](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/github-action-workflows.md)
- Test by merging to your fork's `main` and simulating the scenario
- Create test issues or PRs as evidence and link them in your PR
- Include screenshots of workflow runs where applicable

**Content, styling, or layout changes:**
- Run the dev server and confirm visual output matches expected behavior
- Compare your changes with similar existing pages for consistency
- Check responsive behavior and accessibility

### 🛡️ Quality & Review Standards

Intermediate PRs must be **"working, maintainable, and aligned with project architecture."**

  1. **Working:** The implementation must solve the problem and meet the acceptance criteria.
  2. **Maintainable:** Code should be clear and concise enough for others to understand and debug without your assistance.
  3. **Architectural Alignment:** The solution should fit well with existing component patterns and project conventions.
  4. **Backward Compatibility:** Existing pages and components must not break. 
  5. **Comprehensive Testing:** Must include unit tests covering all new logic paths, edge cases, and failure modes. AI-generated tests based on AI-generated code are grounds for immediate rejection.

**⚠️ Breaking changes**
- Before changing any component API, prop interface, or public-facing layout — stop and check
- If a breaking change is unavoidable: get explicit maintainer approval **before** implementing
- All existing tests should pass as-is.

**🤖 AI notes**
- AI often has out-dated real-world knowledge: it will not be able to identify current packages.
- AI has gaps or incorrect understanding of repo-specific patterns, logic, and component structures.

### ✅ PR Quality Checklist

Before opening your PR, confirm:

- [ ] I have spent the majority of my time researching the problem and solution space extensively, including reviewing relevant code, documentation, and external resources.
- [ ] My implementation works but is also of high quality - including maintainability, readability, and architectural fit.
- [ ] I have checked for breaking changes - no existing pages or components regress.
- [ ] The design fits with the current website's architectural approaches.
- [ ] Every line of code is personally understood and explainable.

Before requesting a review, confirm:
- [ ] I have reviewed the diff line by line.
- [ ] My implementation fully addresses the problem described above.
- [ ] I did not modify files unrelated to this issue
- [ ] Clean git history — no rebase artifacts, merge commits, or unrelated files
- [ ] My commits are signed: `git commit -S -s -m "chore: description"` — [Signing guide](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/signing.md)
- [ ] I have run the following commands locally and **all pass** (CI will automatically fail your PR if any of these fail):
  - `pnpm format:check` — checks code formatting
  - `pnpm lint` — runs the linter (**the CI lint check will block your PR if this fails**)
  - `pnpm test` — runs the test suite
  - `pnpm build` — ensures the project builds successfully
  - [Testing guide](https://github.com/hiero-ledger/hiero-website/blob/main/docs/06-testing-and-quality-checks.md)
- [ ] I have included appropriate tests and all CI checks pass.
- [ ] Double and triple check — intermediate PRs are time-consuming to review

### 📚 Resources & Support

**🆘 Stuck?**
> [!TIP]
> **Comment on this issue:** and describe what you have tried. A maintainer will respond.

**References:**
- [Next.js Documentation](https://nextjs.org/docs)
- [React Documentation](https://react.dev)
- [Tailwind CSS Documentation](https://tailwindcss.com/docs)
- [Vitest Documentation](https://vitest.dev)
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md)
- [Community Calls](https://zoom-lfx.platform.linuxfoundation.org/meetings/hiero?view=week)
- [Discord](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/discord.md)

### PR DESCRIPTION
**Pull request by danielmarv**
# Description
This pull request introduces a major redesign and accessibility upgrade to the site’s `Footer` component, adds a new `BackToTop` control, and improves the `Header`'s scroll behavior. It also removes legacy navigation and border styles from global CSS, and updates test suites to match the new features and structure.

**Footer Redesign and Navigation Overhaul:**

* The `Footer` is completely reimplemented with a new layout, branding, and navigation structure. It now uses shared navigation data, groups links under labeled headings, adds social and affiliation logos, and ensures all external links open safely in new tabs with accessible names. The visual style is updated with gradients, section labels, and improved color contrast. (`src/components/Footer/index.tsx`, `src/components/Footer/__tests__/Footer.test.tsx`) [[1]](diffhunk://#diff-690031c63c881f194e793d09ea7a366a62b3cc37962f1a81fe5396274deeaec4R1-R243) [[2]](diffhunk://#diff-49174bcd61512bcf5999e52b5c49380515dffd02fa0588cafa6bb28d14c5999bL1-R145)
* The legacy snapshot test for the old footer is removed, reflecting the new dynamic structure. (`src/components/Footer/__tests__/__snapshots__/Footer.test.tsx.snap`)

**Back to Top Control:**

* A new `BackToTop` button is implemented and integrated into the footer. It scrolls smoothly to the top unless reduced motion is preferred, and moves keyboard focus to the header for accessibility. Comprehensive tests are added for its behavior. (`src/components/BackToTop/index.tsx`, `src/components/BackToTop/__tests__/BackToTop.test.tsx`) [[1]](diffhunk://#diff-453bb82e1a8e352809542b96f094a16f93299dfe2d8d968ba5c06b8a1ed254d5R1-R43) [[2]](diffhunk://#diff-cc513374d941b05e0a67e9970681d60ed2dfeb605a8398a474fc88ba6fc424e6R1-R61)

**Header Improvements:**

* The `Header` now hides while scrolling down and reappears when scrolling up, improving usability. Tests verify its scroll behavior and event listener cleanup. The visual style is updated to match the new design. (`src/components/Header/__tests__/Header.test.tsx`, `src/components/Header/__tests__/__snapshots__/Header.test.tsx.snap`) [[1]](diffhunk://#diff-b59f7f7b45f3d566cbc585333f16d812adb29f02fa04b126b5f4d2aa9ff47898L1-R17) [[2]](diffhunk://#diff-b59f7f7b45f3d566cbc585333f16d812adb29f02fa04b126b5f4d2aa9ff47898R27-R79) [[3]](diffhunk://#diff-6fdc3b049cd85b1ac578b2d2493aaf92d8d39ef6735f1c21032ff3c2ad9f406dL5-R5)

**Global CSS Cleanup:**

* Removes legacy navigation and border styles from `globals.css`, clarifying that header and footer edges are now styled within their respective components. (`src/app/globals.css`)

These changes collectively modernize the site’s navigation and footer, improve accessibility, and lay the groundwork for a more consistent user experience.

## Changes Made
- [ ] Added...
- [ ] Modified...
- [ ] Fixed...

## Related Issues

Fixes: #474 #475 

## Checklist
- [ ] Tests added/updated
- [ ] Documentation updated
- [ ] Linting passes
- [ ] Branch up-to-date with main

## Summary by CodeRabbit

* **New Features**
  * Added a scroll-aware header that hides while scrolling down and gains enhanced styling when scrolled.
  * Expanded the footer with branding, grouped navigation, social links, affiliations, policies, and a Back to top control.
  * Improved navigation accessibility, including current-page indicators and new-tab descriptions.
* **Bug Fixes**
  * Back to top respects reduced-motion preferences and restores keyboard focus appropriately.
  * Updated repository statistics.
* **Tests**
  * Added coverage for scrolling behavior, navigation links, footer content, accessibility, and responsive menus.

### CODE REVIEW

**Review by aceppaluni — CHANGES_REQUESTED** [review]
@danielmarv Is there no netlify preview of these changes?

**Comment by danielmarv** [comment]
> @danielmarv Is there no netlify preview of these changes?

the netlify deploy preview is only for the prs against the main branck

**Comment by aceppaluni** [comment]
@danielmarv Are you able to add a screenshot of what it looks like when you run it locally? I was wanting to see the preview if possible.

**Review by exploreriii — CHANGES_REQUESTED** [review]
Looks good but there are some bugs
Nav links are invisible between 640px and 767px and linkClass()'s typography
header nav a* rules and .active-navigation from globals.css are probably dead code
social icons missed the sm → md migration
The tests are also fialing, which ci is not picking up since it runs against main (maybe we should add the working branch)

i think we can avoid writing unlayered CSS in globals.css and avoid it leaking into component internals - maybe this can be a separate cleanup pr to prepare for the redesign

we also have lots of things defined in mutliple places which will create bugs

we can add a prettier css plugin and refine the ci to catch things like this 

so i think we can do some prep work befor eadding more pr's

File: src/components/Menu/index.tsx [hunk 1]
```diff
@@ -12,14 +12,206 @@ import {
...
+/** Horizontal extent of one label, measured relative to the list. */
+interface Rule {
+  left: number;
+  width: number;
+}
+
+/**
+ * The spec assigns `red-light` to the active-link underline. Rather than give
+ * each link its own rule, one shared rule slides between them: it marks the
+ * current page at rest and follows the pointer or keyboard focus, which is what
```
Comment (aceppaluni) [inline src/components/Menu/index.tsx:30]: The rule moves on ```onFocus```, but I don't see a  corresponding focus-leave handling. When keyboard focus moves from a nav link to the social links or another element, the rule stays on the previously focused nav item. Is this correct or should we settle/reset it when focus leaves the nav so keyboard interaction matches the pointer behavior?

**Review by aceppaluni — CHANGES_REQUESTED** [review]
One comment from me, the rest looks good :)

File: src/components/Menu/index.tsx (hunk 1 above)
Comment (danielmarv) [reply src/components/Menu/index.tsx:30]: but i think that os fine3 for now can be a good first issue fix

**Review by exploreriii — APPROVED** [review]

**Review by aceppaluni — APPROVED** [review]
